packages/clerk-js/src/ui/ImpersonationFab/ImpersonationFab.tsx

```diff
@@ -117,9 +123,9 @@ export const ImpersonationFab = () => {
           backgroundColor: t.colors.$white,
           fontFamily: t.fonts.$main,
           ':hover #cl-impersonationText': {
-            maxWidth: `calc(50vw - ${eyeWidth} - 2 * ${right})`,
+            maxWidth: `min(calc(50vw - ${eyeWidth} - 2 * ${right}), ${titleLength}ch)`,
```

Comment:
I was pretty much sure you couldn't live with the delayed animation - nicely done ✔️